evm/src/generation/mod.rs

```diff
@@ -216,7 +217,8 @@ pub fn generate_traces<F: RichField + Extendable<D>, const D: usize>(
     PublicValues,
     GenerationOutputs,
 )> {
-    let mut state = GenerationState::<F>::new(inputs.clone(), &KERNEL.code);
+    let mut state = GenerationState::<F>::new(inputs.clone(), &KERNEL.code)
+        .map_err(|_| anyhow!("Failed to parse all the initial prover inputs"))?;
```

Comment:
nit: Could also do
```rust
.map_err(|err| anyhow!("Failed to parse all the initial prover inputs: {:?}", err))?;
```
to get more info on which ProgramError was thrown. Also applies elsewhere in this PR.